svc-events/events/subscriptionCache.go

```diff
@@ -0,0 +1,210 @@
+		subCache := evmodel.Subscription{
+			Id:                   sub.SubscriptionID,
+			Destination:          sub.EventDestination.Destination,
+			EventTypes:           sub.EventDestination.EventTypes,
+			MessageIds:           sub.EventDestination.MessageIds,
+			SubordinateResources: sub.EventDestination.SubordinateResources,
+			ResourceTypes:        sub.EventDestination.ResourceTypes,
+			SubscriptionType:     sub.EventDestination.SubscriptionType,
+			OriginResources:      sub.EventDestination.OriginResources,
+			DeliveryRetryPolicy:  sub.EventDestination.DeliveryRetryPolicy,
+		}
+		addEmptyOriginSubscriptionCache(subCache.Id)
+		subscriptionsCache[subCache.Id] = subCache
+	} else {
+		for _, host := range sub.Hosts {
+			subCache := evmodel.Subscription{
+				Id:                   sub.SubscriptionID,
+				Destination:          sub.EventDestination.Destination,
+				EventTypes:           sub.EventDestination.EventTypes,
+				MessageIds:           sub.EventDestination.MessageIds,
+				SubordinateResources: sub.EventDestination.SubordinateResources,
+				ResourceTypes:        sub.EventDestination.ResourceTypes,
+				SubscriptionType:     sub.EventDestination.SubscriptionType,
+				OriginResources:      sub.EventDestination.OriginResources,
+				DeliveryRetryPolicy:  sub.EventDestination.DeliveryRetryPolicy,
+			}
+			addSubscriptionCache(host, subCache.Id)
+			subscriptionsCache[subCache.Id] = subCache
+		}
+	}
+}
+
+//addSubscriptionCache add subscription in corresponding cache based on key type
+func addSubscriptionCache(key string, subId string) {
+	if strings.Contains(key, "Collection") {
+		data, isExists := collectionToSubscriptionsMap[key]
+		if isExists {
+			data[subId] = true
+			collectionToSubscriptionsMap[key] = data
+		} else {
+			data := make(map[string]bool)
+			data[subId] = true
+			collectionToSubscriptionsMap[key] = data
+		}
+		return
+	} else {
+		_, err := uuid.FromString(key)
+		if err == nil {
+			addAggregateSubscriptionCache(key, subId)
+			return
+		} else {
+			data, isExists := systemToSubscriptionsMap[key]
+			if isExists {
+				data[subId] = true
+				systemToSubscriptionsMap[key] = data
+			} else {
+				data := make(map[string]bool)
+				data[subId] = true
+				systemToSubscriptionsMap[key] = data
+			}
+			return
+		}
+	}
+}
+
+func addEmptyOriginSubscriptionCache(subscriptionId string) {
+	data, isExists := emptyOriginResourceToSubscriptionsMap["0"]
+	if isExists {
+		data[subscriptionId] = true
+		emptyOriginResourceToSubscriptionsMap["0"] = data
+	} else {
+		emptyOriginResourceToSubscriptionsMap["0"] = map[string]bool{subscriptionId: true}
+	}
+}
+func addAggregateSubscriptionCache(key, subId string) {
+	data, isExists := aggregateIdToSubscriptionsMap[key]
+	if isExists {
+		data[subId] = true
+		aggregateIdToSubscriptionsMap[key] = data
+	} else {
+		aggregateIdToSubscriptionsMap[key] = map[string]bool{subId: true}
+	}
+}
+
+func getAllAggregates() {
```

Comment:
This code is duplicated in multiple places.... refactor

Reply:
done